### DISCUSSION TITLE
What does it mean when the lab instructions says we need to "implement sockaddr_in type for sock_addr"

### DISCUSSION THREAD

**Discussion by orangepineapple** [ANLAB-KAIST/KENSv3#146] [Project Q&A 2. Transport Layer (TCP)]
I am reading the instructions for PA2 and I am a bit confused by what it means when it says we have the implement sock_addr_in for sock_adder, based on the comments in the code I assumed we just get a sock_adder as input, is there any reason we cannot just store that object and reference it for when the client asks us for IP in other function calls? 
    // this->syscall_bind(
    //     syscallUUID, pid, std::get<int>(param.params[0]),
    //     static_cast<struct sockaddr *>(std::get<void *>(param.params[1])),
    //     (socklen_t)std::get<int>(param.params[2]));

    //1) we want to store the sockaddr pointer in our hashmap of hashmaps (pid, fd) as the key

**Comment by Pusnow** [comment]
Sorry for the confusion. The sentence is quite misleading.

You can use `struct sockaddr_in` that Linux provides, and you need to *support* only [sockaddr_in](https://linux.die.net/man/7/ip) type for sockaddr.
